// chaik/test/db_test.js
const assert = require('assert')
    , co = require('co')

    , knex = require('knex')

let see = require('../')

let config = {
    client: 'mysql',
    connection: {
        host: '127.0.0.1',
        user: 'root',
        password: '',
        database: 'xxx'
    }
}

// test sql: create table test(`id` int not null, `name` varchar(20) not null, primary key(`id`))

see.initDbConnection(config)

let db = knex(config)

exports.run = function () {
    assert.doesNotThrow(function () {
        co(function* () {
            let tableName = "test"
            yield db.raw('truncate table `test`')

            let entity = { id: 1, name: "xxx" }
            let entity2 = { id: 2, name: "xxx" }
            yield db(tableName).insert(entity)
            yield db(tableName).insert(entity2)

            see.in.database(entity, tableName, { id: 1 })
            see.in.database([entity, entity2], tableName, { id: 1 })
            see.notIn.database(tableName, { id: 2 })
        }).then(function (value) {
            console.log('db test passed')
        }, function (err) {
            console.error(err.stack);
        });
    })
}
